packages/react-core/src/components/Hint/__tests__/Hint.test.tsx

```diff
@@ -1,17 +1,63 @@
 import * as React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import { Hint } from '../Hint';
 import { HintBody } from '../HintBody';
 import { HintTitle } from '../HintTitle';
 import { HintFooter } from '../HintFooter';
 
-test('simple hint', () => {
-  const { asFragment } = render(
-    <Hint>
-      <HintTitle>Title</HintTitle>
-      <HintBody>Body</HintBody>
-      <HintFooter>Footer</HintFooter>
-    </Hint>
-  );
-  expect(asFragment()).toMatchSnapshot();
+describe('Hint', () => {
+  it('renders without children', () => {
+    render(<Hint data-testid='Hint' />);
+
+    expect(screen.getByTestId('Hint')).toBeVisible();
+  });
+
+  it('renders children', () => {
+    render(<Hint>Test</Hint>);
+
+    expect(screen.getByText('Test')).toBeVisible();
+  });
+
+  it('renders with class pf-c-hint', () => {
+    render(<Hint>Test</Hint>);
+
+    const hint = screen.getByText('Test');
+
+    expect(hint).toHaveClass('pf-c-hint');
+  });
+
+  it('renders with custom class names provided via prop', () => {
+    render(<Hint className='custom-classname'>Test</Hint>);
+
+    const hint = screen.getByText('Test');
+
+    expect(hint).toHaveClass('custom-classname');
+  });
+
+  it('does not render actions options when not passed', () => {
+    render(<Hint>Test</Hint>);
+
+    const actions = screen.queryByText('actions');
+
+    expect(actions).not.toBeInTheDocument();
+  });
+
+  it('renders actions options', () => {
+    render(<Hint actions="actions">Test</Hint>);
+
+    const actions = screen.getByText("actions");
+
+    expect(actions).toBeVisible();
+  });
+
+  it('renders hint with body, title, and footer', () => {
+    const { asFragment } = render(
+      <Hint>
+        <HintTitle>Title</HintTitle>
+        <HintBody>Body</HintBody>
+        <HintFooter>Footer</HintFooter>
+      </Hint>
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
```

Comment:
Can you add a test to check for the `pf-c-hint__actions` class being applied to the actions?

Reply:
Also I would like to see a test for each of these components which verifies that other props which aren't explicitly declared are spread properly.

Reply:
It doesn't seem like it's causing problems, but I did notice that in [the core demo without an actions element ](https://www.patternfly.org/v4/components/hint/html#default-with-no-title)the whole actions div actually isn't rendered.

Regardless for now since (currently at least) the actions div always renders there is no need to check if it applies when an `actions` is passed IMO. We just need to test that what *is* passed to `actions` is placed in that div as we expect it to be.